app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/DefaultValues.kt

```diff
@@ -1,10 +1,16 @@
 package dogacel.kotlinx.protobuf.gen
 
 import com.google.protobuf.Descriptors
+import com.google.protobuf.Descriptors.FieldDescriptor
 import com.squareup.kotlinpoet.CodeBlock
 import com.squareup.kotlinpoet.TypeName
 
 object DefaultValues {
+    // Indicate whether the field unconditionally has a value; true if the field is PROTO2 and marked `required`, or if
+    // the field has a default value when unset.
+    private val FieldDescriptor.unconditionallyHasValue: Boolean get() =
+        !isOptional || !options.features.hasFieldPresence()
```

Comment:
I am not sure if this works as intended. 

In proto 3, having field presence = being marked as `optional`.

The method name was a little confusing, maybe we should name that something like `hasFieldPresence()`?

Reply:
Also I wonder whether `optional int32 a = 1` in proto3 and `int32 a = 1` in edition 2023 both have `hasFieldPresence` set to true. Can you cross-check, write a test and confirm?